dlrover/python/master/hyperparams/simple_strategy_generator.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2022 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+from typing import Dict, List
+
+from dlrover.python.common.constants import NodeType
+from dlrover.python.common.grpc import (
+    DataLoaderConfig,
+    OptimizerConfig,
+    ParallelConfig,
+)
+from dlrover.python.common.log import default_logger as logger
+from dlrover.python.common.node import Node
+from dlrover.python.master.hyperparams.strategy_generator import (
+    StrategyGenerator,
+)
+from dlrover.python.master.stats.reporter import JobMeta, LocalStatsReporter
+
+gpu_stats = [
+    {
+        "index": 0,
+        "total_memory_gb": 24,
+        "used_memory_gb": 0,
+    },
+    {
+        "index": 1,
+        "total_memory_gb": 24,
+        "used_memory_gb": 2.631,
+    },
+]
+
+# Define model configuration as a dictionary
+mock_model_config = {
+    "block_size": 128,
+    "n_layer": 6,
+    "n_heads": 6,
+    "n_embd": 384,
+}
+
+mock_dataloader_config = DataLoaderConfig(0, "simple_dataloader", 32, 2, 0)
+
+
+class SimpleStrategyGenerator(StrategyGenerator):
+    """
+    A simple strategy generator for local optimization.
+    This class is responsible for generating optimal configurations for data
+    loaders and optimizers
+    """
+
+    def __init__(self, job_uuid):
+        self._job_uuid = job_uuid
+        self._stats_collector = LocalStatsReporter(JobMeta(job_uuid))
+
+    def start(self):
+        try:
+            logger.info("Start local strategy generator")
+            threading.Thread(
+                target=self._extract_node_resource,
```

Comment:
Why do we need to start a thread to execute `_extract_node_resource`?

Reply:
Not necessary. This is a leftover during development time.